File: src/main/java/discretemath/bool/operator/FunctionallyCompleteOperatorsSet.java
package discretemath.bool.operator;

import java.util.Set;

import static discretemath.bool.operator.Operators.*;


public enum FunctionallyCompleteOperatorsSet {

	AND_OR_NOT(AND, OR, NOT), AND_NOT(AND, NOT), OR_NOT(OR, NOT), SINGLE_NAND(NAND), SINGLE_NOR(NOR);

	private final Set<Operator> operators;

	FunctionallyCompleteOperatorsSet(Operator... operators) {
		this.operators = Set.of(operators);
	}

	public boolean contains(Operator operator) {
		return operators.contains(operator);
	}
}

File: src/main/scala/discretemath/bool/operator/FunctionallyCompleteOperatorsSet.scala
package discretemath.bool.operator

import discretemath.bool.operator.Operators._;

sealed abstract class FunctionallyCompleteOperatorsSet(operators: Operator*) {

  def contains(operator: Operator): Boolean = operators.contains(operator)
}

object FunctionallyCompleteOperatorsSet {

  case object AND_OR_NOT extends FunctionallyCompleteOperatorsSet(AND, OR, NOT)
  case object AND_NOT extends FunctionallyCompleteOperatorsSet(AND, NOT)
  case object OR_NOT extends FunctionallyCompleteOperatorsSet(OR, NOT)
  case object ONLY_NAND extends FunctionallyCompleteOperatorsSet(NAND)
  case object ONLY_NOR extends FunctionallyCompleteOperatorsSet(NOR)
}


